js/app/dashboard/components/Tour.js

```diff
@@ -1,45 +1,184 @@
-import React from 'react'
+import React, {useState, useEffect} from 'react'
+import { connect } from 'react-redux'
 import { withTranslation, useTranslation } from 'react-i18next'
+import { Draggable, Droppable } from "react-beautiful-dnd"
+import _ from 'lodash'
+import Popconfirm from 'antd/lib/popconfirm'
+
 import Task from './Task'
+import { removeTaskFromTour, modifyTour, deleteTour, unassignTasks } from '../redux/actions'
+import { isTourAssigned, tourIsAssignedTo } from '../../../shared/src/logistics/redux/selectors'
+import classNames from 'classnames'
+import { useContextMenu } from 'react-contexify'
+import { getDroppableListStyle } from '../utils'
+
+
+const Tour = ({ tour, removeTaskFromTour, unassignTasks, isDroppable, modifyTour, deleteTour }) => {
 
-const Tour = ({ tour, tasks, username = null, unassignTasks = null }) => {
 
-  const { t } = useTranslation()
+  const { t } = useTranslation(),
+        { show } = useContextMenu({
+          id: 'dashboard',
+        }),
+        collapseId = `tour-panel-${tour['@id'].replaceAll('/', '-')}`,
+        [toggleInputForName, setToggleInputForName] = useState(false),
+        [tourName, setTourName] = useState(tour.name),
+        onEditSubmitted = async (e) => {
+          e.preventDefault()
+          $('.task__draggable').LoadingOverlay('show', {image: false})
```

Comment:
yes definitely, it is a bad habit of the old me :D I have changed this code on Friday in order to use redux state to control tour panel (was necessary because of refresh), I will cherry pick this commit

Reply:
Ideally, yes. Not sure if we get time for this, though. For now, I would at least avoid using it in the new code